src/ConformalTracking.cc

```diff
@@ -1090,34 +1148,47 @@ void ConformalTracking::buildNewTracks(UniqueKDTracks& conformalTracks, SharedKD
             // Otherwise replace the existing track
             conformalTrack = std::move(bestTrack);
             bestTrackUsed  = true;
+            streamlog_out(DEBUG9) << "- New = existing + segment. Replaced existing with new" << std::endl;
           }
 
           // If the new track is a subtrack of an existing track, don't consider it further (already try removing bad hits from tracks
-          else if (nOverlappingHits == bestTrack->m_clusters.size())
-            break;
 
+          else if (nOverlappingHits == bestTrack->m_clusters.size()) {
+            streamlog_out(DEBUG9) << "- New + segment = existing. New ignored" << std::endl;
+            break;
+          }
           // Otherwise take the longest
           else if (bestTrack->m_clusters.size() >= conformalTrack->m_clusters.size()) {  // New track longer/equal in length
-
             // Take it
             conformalTrack = std::move(bestTrack);
             bestTrackUsed  = true;
+            streamlog_out(DEBUG9) << "- New longer/equal. Replaced existing with new" << std::endl;
 
-          } else if (bestTrack->m_clusters.size() < conformalTrack->m_clusters.size())  // Old track longer
+          } else if (bestTrack->m_clusters.size() < conformalTrack->m_clusters.size()) {  // Old track longer
+            streamlog_out(DEBUG9) << "- Old longer. New ignored" << std::endl;
             break;
-
+          }
           break;
         }
       }
 
       // If not a clone, save the new track
       if (!clone) {
         bestTrackUsed = true;
+
+        streamlog_out(DEBUG9) << "- Track is not a clone. Pushing back best track with chi2/ndof " << bestTrack->chi2ndof()
+                              << " and hits: " << std::endl;
+
         if (debugSeed && kdhit == debugSeed) {
           streamlog_out(DEBUG7) << "== Pushing back best track with chi2/ndof " << bestTrack->chi2ndof() << std::endl;
         } else {
           streamlog_out(DEBUG7) << "Pushing back best track with chi2/ndof " << bestTrack->chi2ndof() << std::endl;
         }
+
+        for (unsigned int cluster = 0; cluster < bestTrack->m_clusters.size(); cluster++) {
```

Comment:
inside the loop condition
`... .size() && streamlog::out.write< DEBUG9 >(); ...`